velox/functions/sparksql/Map.cpp

```diff
@@ -96,21 +98,27 @@ class MapFunction : public exec::VectorFunction {
     auto offsets = mapResult->mutableOffsets(rows.end());
     auto rawOffsets = offsets->asMutable<int32_t>();
 
+    // Setting keys and value elements
+    auto keysResult = mapResult->mapKeys();
+    auto valuesResult = mapResult->mapValues();
+    auto baseOffset = keysResult->size();
+    VELOX_CHECK(
+        baseOffset == valuesResult->size(),
+        "Map keys size must equal values size.");
+
     // Setting size and offsets
+    vector_size_t offset = baseOffset;
     rows.applyToSelected([&](vector_size_t row) {
       rawSizes[row] = mapSize;
-      rawOffsets[row] = row * mapSize;
+      rawOffsets[row] = offset;
+      offset += mapSize;
     });
 
-    // Setting keys and value elements
-    auto keysResult = mapResult->mapKeys();
-    auto valuesResult = mapResult->mapValues();
-    const auto resultSize = rows.end() * mapSize;
+    const auto resultSize = baseOffset + rows.countSelected() * mapSize;
```

Comment:
Here resultSize is the same as offset. Hence, you can just use 'offset'.

Reply:
Yes, updated, thanks.